gatsby/src/pages/search.tsx

```diff
@@ -1,14 +1,309 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
+import { Link } from "gatsby";
+import { useLocation } from "@reach/router";
+import { Card } from "@nice-digital/nds-card";
+import { SimplePagination } from "@nice-digital/nds-simple-pagination";
+import { PageHeader } from "@nice-digital/nds-page-header";
+
 import { Layout } from "../components/Layout/Layout";
 import { SEO } from "../components/SEO/SEO";
+import { setTimeout } from "timers";
+
+interface SearchResults {
+	failed: boolean;
+	resultCount: number;
+	firstResult: number;
+	lastResult: number;
+	finalSearchText: string;
+	finalSearchTextNoStopWords: string;
+	pageSize: number;
+	originalSearch: {
+		searchText: string;
+	};
+	pagerLinks: {
+		previous: {
+			fullUrl: string;
+		} | null;
+		next: {
+			fullUrl: string;
+		} | null;
+		pages: {
+			title: string;
+			url: {
+				fullUrl: string;
+			};
+			isCurrent: boolean;
+		}[];
+	};
+	documents: Document[];
+}
+
+type Document = {
+	id: string;
+	metaDescription: string;
+	contentId: string;
+	pathAndQuery: string;
+	sourceUrl: string;
+	teaser: string;
+	title: string;
+	url: string;
+};
+
+function titleString(searchText: string, pageIndex: number): string {
+	return [pageIndex > 1 && `Page ${pageIndex}`, searchText, "Search results"]
+		.filter(Boolean)
+		.join(" | ");
+}
 
 const SearchPage: React.FC = () => {
+	const [data, setData] = useState<SearchResults | null>(null);
+	const [error, setError] = useState<boolean>(false);
+	const [a11yMessage, setA11yMessage] = useState<string>("");
+
+	function announce(message: string): void {
```

Comment:
This function seems a bit redundant, but if we're keeping it should we memoize it?

Reply:
I think this probably needs a timeout to reset it to blank text content. Otherwise having two 'search results loaded' in a row won't announce anything...?